postgre-client/src/asset_filter_client.rs

```diff
@@ -103,19 +103,53 @@ impl PgClient {
         }
         // Add GROUP BY clause if necessary
         if group_clause_required {
-            query_builder.push(" GROUP BY assets_v3.ast_pubkey, assets_v3.ast_slot_created, assets_v3.ast_slot_updated ");
+            query_builder.push(" GROUP BY ast_pubkey, ast_slot_created, ast_slot_updated ");
         }
+        query_builder.push(") ");
+
+        // the function with a side-effect that mutates the query_builder
+        if let Some(owner_address) = &filter.owner_address {
+            if let Some(ref token_type) = filter.token_type {
+                match *token_type {
+                    TokenType::Fungible
+                    | TokenType::NonFungible
+                    | TokenType::RegularNFT
+                    | TokenType::CompressedNFT => {}
+                    TokenType::All => {
+                        query_builder.push("  UNION ALL ");
```

Comment:
Do you need `union all` here? It might return you duplicate asset_ids, which doesn't look like expected behaviour. Although to be fair it should not receive duplicates in this union, I'd suggest using just the `UNION ` here

Reply:
I do not.